### PR TITLE
D3D: Scaling on window resizing

### PR DESCRIPTION
**Pull request by MarcelHB**
## Description

Some findings:
* `SDL_RENDER_TARGETS_RESET` is exclusively fired for D3D9 (or `direct3d` backend), D3D11 is unaffected at all. Anyway, unlike the removed comment, the [SDL docs imply](https://wiki.libsdl.org/SDL_EventType) that existing textures can be reused. So I think it's the easiest way just to make every window redraw on the next occasion.
* `SDL_RENDER_DEVICE_RESET` is exlusively fired for D3D11 and is not so much related to the issue. From what I read in [MSDN](https://docs.microsoft.com/en-us/windows/uwp/gaming/handling-device-lost-scenarios), it covers some very, very edgy cases that I'd like to ignore here.

See #1317

## Checklist

- [x] Commit messages are descriptive and explain the rationale for changes
- [x] I used the same coding style as the surrounding code 
- [x] I have tested the proposed changes
- [x] I extended the documentation, if necessary
- [x] The proposed change builds also on our build bots (check after submission)

### CODE REVIEW

**Comment by bradallred** [comment]
excellent. thank you.

a couple of things:

1. I think this could also fix #1296 if we do this for `SDL_APP_DIDENTERFOREGROUND` as well
2. we dont need to make `Interface` a middle man to everything. Just grab the `WindowManager` and call on it directly.

**Comment by MarcelHB** [comment]
1. Ok, I'll try to have a look.
2. Had it like this first, but then I noticed nobody ever `GEM_EXPORT`ed the WM so far, which isn't so bad from a coupling perspective.

**Comment by bradallred** [comment]
im saying `Interface` already has `GetWindowManager`

**Comment by bradallred** [comment]
if you are concerned about coupling you can always just create a new "application" event type in the `EventManager` and let window manager handle all this internally.

**Comment by MarcelHB** [comment]
I spotted it, but the linker complained when being used outside of core.

The event design sounds better, I'll check it out.

File: gemrb/core/Interface.cpp [hunk 1]
```diff
@@ -1849,6 +1850,16 @@ int Interface::Init(InterfaceConfig* config)
...
+	EventMgr::EventCallback appEventCallback = [this](const Event& e) {
+		if (e.type == Event::EventType::RedrawRequest) {
+			winmgr->MarkAllDirty();
+		}
+
+		return true;
+	};
+
+	appEventMonitor = EventMgr::RegisterEventMonitor(appEventCallback, Event::EventTypeMask::AllApplicationMask);
+
```
Comment (bradallred) [inline gemrb/core/Interface.cpp:1862]: you dont need to do this. Just handle the event in `WindowManager::DispatchEvent`

**Comment by bradallred** [comment]
beautiful. thank you.

**Pull request merged**